lupa/_lupa.pyx

```diff
@@ -1035,11 +1034,8 @@ cdef class _LuaIter:
             if self._obj is None:
                 raise StopIteration
             # iterable object
-            lua.lua_rawgeti(L, lua.LUA_REGISTRYINDEX, self._obj._ref)
+            self._obj.push_lua_object(L)
             if not lua.lua_istable(L, -1):
-                if lua.lua_isnil(L, -1):
```

Comment:
Note that `_LuaObject.push_lua_object` already checks whether the entry in the registry is `nil`.